libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -557,15 +557,72 @@ - (void)beginUserAgentFlow {
         [approvalUrl appendString:scopeString];
     }
     
+    // JWT Flow
+    if (self.credentials.jwt && self.credentials.instanceUrl) {
+        [self swapJWTWithcompletionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+            bool swapOK = NO;
+            if (!error) {
+                NSError *jsonError = nil;
+                id json = nil;
+                
+                json = [NSJSONSerialization JSONObjectWithData:data options:0 error:&jsonError];
+                if (nil == jsonError && [json isKindOfClass:[NSDictionary class]]) {
+                    NSDictionary *dict = (NSDictionary *)json;
+                    if (dict[kSFOAuthAccessToken]) {
+                        NSString *escapedString = [self URLEncodeStringFromString:approvalUrl];
+                        NSString* approvalUrl = [NSString stringWithFormat:@"%@://%@/secur/frontdoor.jsp?sid=%@&retURL=%@", self.credentials.protocol, self.credentials.instanceUrl, dict[kSFOAuthAccessToken],escapedString];
+                        [self doLoadURL:approvalUrl withCookie:YES];
+                        swapOK = YES;
+                        self.credentials.jwt = nil;
+                    }
+                }
+            }
+            if (!swapOK) {
+                [self log:SFLogLevelInfo msg:@"Fail to swap JWT for access token"];
+                [self doLoadURL:approvalUrl withCookie:NO];
+            }
+        }];
+    }
+    else {
+        [self doLoadURL:approvalUrl withCookie:NO];
+    }
+}
+
+
+- (void)swapJWTWithcompletionHandler:(void (^)(NSData *data, NSURLResponse *response, NSError *error))completionHandler  {
+    NSString *url = [[NSString alloc] initWithFormat:@"%@://%@%@", self.credentials.protocol,
+                     self.credentials.instanceUrl.absoluteString,
+                     kSFOAuthEndPointToken];
+    
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:url] cachePolicy:NSURLRequestReloadIgnoringCacheData
+                                                       timeoutInterval:self.timeout];
+    NSString *bodyStr = [NSString stringWithFormat:@"grant_type=urn%%3Aietf%%3Aparams%%3Aoauth%%3Agrant-type%%3Ajwt-bearer&assertion=%@",self.credentials.jwt];
```

Comment:
Can we create this by instantiating the unencoded string, then encoding it?  The original string will be easier to read/maintain.